src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -2779,15 +2775,21 @@ public void visitMethodDef(JCMethodDecl tree) {
                 syms.methodClass);
         }
 
+        Type prevRestype = currentRestype;
         JCMethodDecl prevMethodDef = currentMethodDef;
         MethodSymbol prevMethodSym = currentMethodSym;
+        int prevVariableIndex = variableIndex;
```

Comment:
This machinery is needed to generate "stable" local variable names, so that lambda deduplication will work as expected. This code is borrowed from `TransPatterns`.